rs/pocket_ic_tests/src/lib.rs

```diff
@@ -0,0 +1,347 @@
+};
+use std::time::{Duration, SystemTime};
+
+#[derive(Default)]
+pub struct StateMachine {
+    sm: ic_state_machine_tests::StateMachine,
+}
+
+#[derive(Default)]
+pub struct StateMachineBuilder {
+    sm_builder: ic_state_machine_tests::StateMachineBuilder,
+}
+
+impl StateMachineBuilder {
+    // Tricky!!!
+    // subnet_config.scheduler_config.max_instructions_per_slice = MAX_INSTRUCTIONS_PER_SLICE
+    // rate_limiting_of_instructions: FlagStatus::Disabled
+    pub fn with_config(self, config: Option<StateMachineConfig>) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_config(config),
+        }
+    }
+    // need to make subnet IDs configurable for all subnet kinds in PocketIC
+    pub fn with_routing_table(self, routing_table: RoutingTable) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_routing_table(routing_table),
+        }
+    }
+
+    // can be refactored into specifying the desired PocketIC subnet kind, e.g., II or fiduciary
+    pub fn with_subnet_type(self, subnet_type: SubnetType) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_subnet_type(subnet_type),
+        }
+    }
+    pub fn with_subnet_size(self, subnet_size: usize) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_subnet_size(subnet_size),
+        }
+    }
+    pub fn with_extra_canister_range(self, id_range: std::ops::RangeInclusive<CanisterId>) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_extra_canister_range(id_range),
+        }
+    }
+
+    pub fn new() -> Self {
+        Self {
+            sm_builder: ic_state_machine_tests::StateMachineBuilder::new(),
+        }
+    }
+
+    pub fn with_state_machine_state_dir(self, state_dir: Box<dyn StateMachineStateDir>) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_state_machine_state_dir(state_dir),
+        }
+    }
+
+    pub fn with_time(self, time: Time) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_time(time),
+        }
+    }
+
+    pub fn with_current_time(self) -> Self {
+        let time = Time::try_from(SystemTime::now()).expect("Current time conversion failed");
+        self.with_time(time)
+    }
+
+    pub fn with_nns_subnet_id(self, nns_subnet_id: SubnetId) -> Self {
+        Self {
+            sm_builder: self.sm_builder.with_nns_subnet_id(nns_subnet_id),
+        }
+    }
+
+    pub fn build(self) -> StateMachine {
+        StateMachine {
+            sm: self.sm_builder.build(),
+        }
+    }
+}
+
+impl StateMachine {
+    // tricky!!!
+    pub fn run_until_completion(&self, max_ticks: usize) {
+        self.sm.run_until_completion(max_ticks)
+    }
+    // tricky!!!
+    pub fn num_running_canisters(&self) -> u64 {
+        self.sm.num_running_canisters()
+    }
+    // tricky!!
+    pub fn canister_memory_usage_bytes(&self) -> u64 {
+        self.sm.canister_memory_usage_bytes()
+    }
```

Comment:
can be dropped after refactoring rs/rust_canisters/canister_creator/test/test.rs to use canister status instead (and decreasing the number of created canisters to much less than 1,000 so that canister status can be quickly evaluated for all the canisters)